storage/google/cloud/storage/blob.py

```diff
@@ -223,6 +223,14 @@ def client(self):
         return self.bucket.client
 
     @property
+    def user_project(self):
+        """Project ID used for API requests made via this blob.
+
+        Derived from bucket's value.
```

Comment:
`Blob` instances *must* have a valid bucket (we have lots of other properties which rely on that).